cmd/clusterctl/client/tree/discovery.go

```diff
@@ -19,11 +19,16 @@ package tree
 import (
 	"context"
 
+	"github.com/Azure/go-autorest/autorest/to"
```

Comment:
Looks like this package is only imported for the `StringPtr` function. Can you please switch over to using `"k8s.io/utils/pointer.StringPtr` function? This will save us from having to import a new dependency in out go.mod.